### ISSUE TITLE
Multi battle allies can't use mega evolution

### ISSUE THREAD

**Issue by BuffelSaft** [rh-hideout/pokeemerald-expansion#1247]
This was tested using the custom multibattle feature in the battle engine; I gave the friendly NPC a Metagross holding a Metagrossite, but it didn't mega evolve. Both opponents were able to mega evolve their Pokemon, so this only affects the player's ally.

**Comment by Pyredrid** [comment]
So I've spent a while trying to find this issue, but the whole multi-battle system is a pain.  Do you have a demo script that triggers the battle situation you described?  It'd help greatly with fixing this issue.

**Comment by BuffelSaft** [comment]
Sure, the script I used is the Mossdeep multi-battle, but tweaked to use the battle engine's special instead of the usual one. It's in this [file](https://github.com/BuffelSaft/pokeemerald/blob/master/data/maps/MossdeepCity_SpaceCenter_2F/scripts.inc), the script is named "MossdeepCity_SpaceCenter_2F_EventScript_Steven".

Unfortunately it's full of movements and things that won't just work if you paste it into another map, so it'd still be a pain to deal with. Also, two of the trainers used in it don't exist in the base game, so you'd have to replace them and edit the ally trainer's team.

Might be best if I set up a proper demo script later this week in case this turns out to be a wild goose chase?

**Comment by Pyredrid** [comment]
Yea, from what I can tell, that Steven multi-battle has a bunch of special cases.  A demo script to reproduce the issue would be better.

**Comment by BuffelSaft** [comment]
Retested with a demo script with only the RHH branches:
![Sidneys](https://user-images.githubusercontent.com/28769716/101268281-1c226300-37c6-11eb-917c-92ae52b6d9dc.gif)
All three NPC trainers are Sidney, so the allied Metagross should be able to mega evolve.
Script used:

```
OldaleTown_EventScript_MultiTest::
    choose_mons
    compare VAR_RESULT, 0
    goto_if_eq OldaleTown_EventScript_MultiTestEnd
    multi_2_vs_2 TRAINER_SIDNEY, EverGrandeCity_SidneysRoom_Text_Defeat, TRAINER_SIDNEY, EverGrandeCity_SidneysRoom_Text_Defeat, TRAINER_SIDNEY, TRAINER_BACK_PIC_STEVEN
	specialvar VAR_RESULT, GetBattleOutcome
	compare VAR_RESULT, B_OUTCOME_LOST
	goto_if_eq OldaleTown_EventScript_MultiTest_Lost
	end

OldaleTown_EventScript_MultiTest_Lost::
	fadescreen FADE_TO_BLACK
	special SetCB2WhiteOut
	waitstate

OldaleTown_EventScript_MultiTestEnd::
	end
```
This can be pasted onto any old NPC for testing, the only other changes needed are giving Sidney a Metagross with a Metagrossite, and uncommenting Metagross' mega evolution in src/data/pokemon/evolution.h.

**Comment by NamNguyenUTSA** [comment]
This one is an easy fix!

`In src/battle_controller_player_partner.c` find the function `PlayerPartnerHandleChooseMove(void)` and change this:

`BtlController_EmitTwoReturnValues(1, 10, chosenMoveId | (gBattlerTarget << 8));` 

to the following:

```
if (CanMegaEvolve(gActiveBattler)) // If opponent can mega evolve, do it.
                    BtlController_EmitTwoReturnValues(1, 10, (chosenMoveId) | (RET_MEGA_EVOLUTION) | (gBattlerTarget << 8));
else
                    BtlController_EmitTwoReturnValues(1, 10, (chosenMoveId) | (gBattlerTarget << 8));
```

NPC partners will now mega evolve as long as they've got a mega stone!

**Comment by LOuroboros** [comment]
> 
> 
> This one is an easy fix!
> 
> `In src/battle_controller_player_partner.c` find the function `PlayerPartnerHandleChooseMove(void)` and change this:
> 
> `BtlController_EmitTwoReturnValues(1, 10, chosenMoveId | (gBattlerTarget << 8));`
> 
> to the following:
> 
> ```
> if (CanMegaEvolve(gActiveBattler)) // If opponent can mega evolve, do it.
>                     BtlController_EmitTwoReturnValues(1, 10, (chosenMoveId) | (RET_MEGA_EVOLUTION) | (gBattlerTarget << 8));
> else
>                     BtlController_EmitTwoReturnValues(1, 10, (chosenMoveId) | (gBattlerTarget << 8));
> ```
> 
> NPC partners will now mega evolve as long as they've got a mega stone!

If you already tested the solution, it would be nice if you submitted it as a PR th the battle_engine branch 👍

**Comment by ultima-soul** [comment]
Resolved in #1281.
